client/src/components/ShoeShowPage.js

```diff
@@ -1,43 +1,108 @@
 import React, { useState, useEffect } from "react"
+import { Redirect } from "react-router-dom"
+import ReviewList from "./ReviewList.js"
+import ReviewForm from "./ReviewForm.js"
+import translateServerErrors from "../services/translateServerErrors.js"
 
 const ShoeShowPage = props => {
-    const [shoe, setShoe] = useState({})
+
+    const { user } = props
+    const [shoe, setShoe] = useState({
+      reviews: []
+    })
+    const [shouldRedirect, setShouldRedirect] = useState(false)
+    const [errors, setErrors] = useState({})
 
     const id = props.match.params.id
     const getShoe = async () => {
         try{
             const response = await fetch(`/api/v1/shoes/${id}`)
             if (!response.ok) {
+              if (response.status===422) {
+                const body = await response.json()
+                const newErrors = translateServerErrors(body.errors)
+                return setErrors(newErrors)
+
+              } else {
                 const errorMessage = `${response.status} (${response.statusText})`
                 const error = new Error(errorMessage)
-                throw(error)
+                throw error
               }
-            const shoeData = await response.json()
-            setShoe(shoeData.shoe)
-        }   catch(err) {
-            console.error(`Error in fetch: ${err.message}`)
+            } else {
+              const shoeData = await response.json()
+              setShoe(shoeData.shoe)
+            }
+        } catch(err) {
+          console.error(`Error in fetch: ${err.message}`)
         }
     }
 
-
-    useEffect(() => {
+    const postReview = async (newReviewData) => {
+      try {
+        const response = await fetch(`/api/v1/shoes/${id}/reviews`, {
+          method: "POST",
+          headers: new Headers({
+            "Content-Type": "application/json"
+          }),
+          body: JSON.stringify(newReviewData)
+        })
+        if (!response.ok) {
+          if (response.status === 422) {
+            const body = await response.json()
+            const newErrors = translateServerErrors(body.errors)
+            return setErrors(newErrors)
+          } else {
+            const errorMessage = `${response.status} (${response.statusText})`
+            const error = new Error(errorMessage)
+            throw error
+          }
+        } else {
+          const body = await response.json()
+          const updatedReviews = [...shoe.reviews, body.review]
+          setShoe({ ...shoe, reviews:updatedReviews})
+          setShouldRedirect(true)          
```

Comment:
we have to choose either line 61, or line 63. 

Since we are _staying_ on the page, we shouldnt redirect. 

Make this all work without a redirect.